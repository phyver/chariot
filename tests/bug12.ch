:set show_tuples false
:set verbose 1
:set allow_structs true

data nat where Zero : nat | Succ : nat -> nat

val a = b
and b = c
and c = d
and d = 0

val e = 0
and f = e
and g = f
and h = f

:show functions

