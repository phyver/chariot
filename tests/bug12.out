>> option verbose is set to 1
>> option show_tuples is set to false
>> option verbose is set to 1
>> option allow_structs is set to true
>> inductive type nat was succesfully defined
>> Typing for d, c, b, a successful
>> the definition for a is complete
>> the definition for b is complete
>> the definition for c is complete
>> the definition for d is complete
>> the definitions d, c, b, a are provably correct

>> Typing for h, g, f, e successful
>> the definition for e is complete
>> the definition for f is complete
>> the definition for g is complete
>> the definition for h is complete
>> the definitions h, g, f, e are provably correct

>> functions in environment:
val
   a : nat
    | a = b
and
   b : nat
    | b = c
and
   c : nat
    | c = d
and
   d : nat
    | d = 0

val
   e : nat
    | e = 0
and
   f : nat
    | f = e
and
   g : nat
    | g = f
and
   h : nat
    | h = f

